### ISSUE TITLE
Text doesn't respect requested High Contrast colors

### ISSUE THREAD

**Issue by MikeHillberg** [microsoft/microsoft-ui-xaml#2031]
**Describe the bug**
Xaml text has a feature where it ensures text is high contrast when the user has enabled high contrast mode. It does this by ignoring any Foreground color that's been specified for the text, and by adding a fixed background.

You can disable this behavior with the [UIElement.HighContrastAdjustment](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.UIElement.HighContrastAdjustment), and that page has more information on the feature.

The problem is that (when the feature isn't disabled) it doesn't use the high contrast colors that the user has specified in Settings.

**Steps to reproduce the bug**

* Create a page with:

```
<Button>Click</Button>
<Button HighContrastAdjustment="None">Clack</Button>
```

* Put the PC into high contrast mode
* Run the app and compare the buttons to the settings in the 'High Contrast' page of the Settings app.

**Expected behavior**

Both buttons should match the Settings configuration, but only the second one does.

**Screenshots**

![image](https://user-images.githubusercontent.com/18429489/75365246-c5205280-5871-11ea-920f-cae22b61351d.png)

**Version Info**

Windows 1903
